core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -4292,6 +4293,21 @@ private Sql checkSubQuery(String sql) {
     checkPlanning(program, sql);
   }
 
+  @Test public void testFilterAndProjectWithMultiJoin() throws Exception {
+    final FilterMultiJoinMergeRule filterMultiJoinMergeRule =
+        new FilterMultiJoinMergeRule(LogicalFilter.class, RelFactories.LOGICAL_BUILDER);
+    final ProjectMultiJoinMergeRule projectMultiJoinMergeRule =
+        new ProjectMultiJoinMergeRule(LogicalProject.class, RelFactories.LOGICAL_BUILDER);
```

Comment:
Well, LogicalProject was the default class, so this test does not seem to validate anything new. Does it just duplicate the existing tests?

Reply:
I added a new constructor in FilterMultiJoinMergeRule and ProjectMultiJoinMergeRule that take a generic Filter.class and Project.class as arguments respectively. The unit test just uses the new constructors with LogicalFilter and LogicalProject

Reply:
Assume the implementation is "ignore the parameter and always use `LogicalProject.class`"
Does the test catch that?